go/ct/spc/specification.go

```diff
@@ -1710,6 +1731,76 @@ func logOp(n int) []Rule {
 	return rules
 }
 
+func nonStaticSelfDestOp(revision Revision, warm bool, destinationColdCost, accountCreationFee, refundGas vm.Gas) []Rule {
+
+	var targetWarm Condition
+	var warmColdString string
+	if warm {
+		warmColdString = "warm"
+		targetWarm = IsAddressWarm(Param(0))
+	} else {
+		warmColdString = "cold"
+		targetWarm = IsAddressCold(Param(0))
+	}
+
+	name := fmt.Sprintf("_%v_%v", strings.ToLower(revision.String()), warmColdString)
+
+	inst := instruction{
+		op:        SELFDESTRUCT,
+		name:      name,
+		staticGas: 5000,
+		pops:      1,
+		conditions: []Condition{
+			Eq(ReadOnly(), false),
+			IsRevision(revision),
+			HasNotSelfDestructed(),
+			targetWarm,
+		},
+		parameters: []Parameter{AddressParameter{}},
+		effect: func(s *st.State) {
+			selfDestructEffect(s, destinationColdCost, accountCreationFee, refundGas)
+		},
+	}
+
+	return rulesFor(inst)
+}
+
+func selfDestructEffect(s *st.State, destinationColdCost, accountCreationFee, refundGas vm.Gas) {
+	// Behavior pre cancun: the current account is registered to be destroyed, and will be at the end of the current
+	// transaction. The transfer of the current balance to the given account cannot fail. In particular,
+	// the destination account code (if any) is not executed, or, if the account does not exist, the
+	// balance is still added to the given address.
+
+	// account to send the current balance to
+	destinationAccount := s.Stack.Pop().Bytes20be()
+	currentAccount := s.CallContext.AccountAddress
+	CurrentBalance := s.Accounts.GetBalance(currentAccount)
+
+	dynamicCost := vm.Gas(0)
+	if !CurrentBalance.IsZero() && !s.Accounts.Exist(destinationAccount) {
+		dynamicCost += accountCreationFee
+	}
+
+	dynamicCost += destinationColdCost
+
+	if s.Gas < dynamicCost {
+		s.Status = st.Failed
+		return
+	}
+	s.Gas -= dynamicCost
+	if s.Revision > R07_Istanbul {
+		s.Accounts.MarkWarm(destinationAccount)
+	}
+	// add beneficiary to list in state
+	s.HasSelfDestructed = true
+	s.SelfDestructedJournal = append(s.SelfDestructedJournal, st.NewSelfDestructEntry(s.CallContext.AccountAddress, CurrentBalance.Bytes32be()))
+	s.Status = st.Stopped
+
+	// PC should not increase, but rulesFor does it for all regular cases, so we counter it here.
+	s.Pc--
```

Comment:
nit: not really a good reason. I would remove it and only list the bare minimum of effects.